File: silence/tests/contract.js
import { readFileSync } from 'fs';
import { fileURLToPath } from 'url';
import { join, dirname } from 'path';

const __dirname = dirname(fileURLToPath(import.meta.url));
const CONTRACT = JSON.parse(
  readFileSync(join(__dirname, '../../contracts/api.json'), 'utf8')
);

export const AUTH = CONTRACT.auth;
export const RAWG = CONTRACT.rawg;

export function mockResponse(entry, body = {}) {
  return {
    ok: entry.status < 400,
    status: entry.status,
    json: async () => body,
  };
}

File: corsproxy/tests/contract.py
import json
from pathlib import Path

_path = Path(__file__).resolve().parents[2] / 'contracts' / 'api.json'
CONTRACT = json.loads(_path.read_text())
AUTH  = CONTRACT['auth']
IGDB  = CONTRACT['igdb']


def assert_contract(response, entry):
    assert response.status_code == entry['status'], \
        f"Expected HTTP {entry['status']}, got {response.status_code}"
    for field in entry.get('fields', []):
        body = response.get_json()
        assert field in body, f"Missing '{field}' in response: {body}"
